.github/workflows/test.yml

```diff
@@ -101,11 +55,8 @@ jobs:
         with:
           python-version: ${{ matrix.python }}
 
-      - name: Install dependencies
-        run: pip install -r requirements.txt coveralls
-
-      - name: Install project
-        run: pip install .
+      - name: Install project & dependencies
+        run: pip install .[dev]
```

Comment:
How does this work, how is `.[dev]` resolved?

Reply:
It is part of `setuptools` used to define optional requirements for the project. It reads this part from the `setup.cfg` file.

```ini
[options.extras_require]
dev =
    black
    isort
    pre-commit
    coveralls
```

 In this case we use it to store and install the development dependencies.

For more info have a look: https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies

If you want we can completely ditch the `setup.cfg` file and move everything to `pyproject.toml`